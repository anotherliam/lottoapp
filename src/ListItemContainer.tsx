import styled from "@emotion/native";

/**
 * List Item for both draws and results
 */

interface ListItemContainerProps { refreshing: boolean; }
const ListItemContainer = styled.View(({ refreshing }: ListItemContainerProps) => ({
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    marginVertical: 8,
    marginHorizontal: 12,
    padding: 8,
    backgroundColor: "#eee",
    elevation: 4,
    opacity: refreshing ? 0.5 : 1.0
}));

export default ListItemContainer;