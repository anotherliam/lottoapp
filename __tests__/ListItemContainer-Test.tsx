import "react-native";
import React from "react";
import ListItemContainer from "../src/ListItemContainer";

import renderer from "react-test-renderer";


// Snapshot tests

it("renders and matches snapshot", () => {
    const tree = renderer.create(<ListItemContainer refreshing={false} />).toJSON();
    expect(tree).toMatchSnapshot();
})

it("renders and matches snapshot when refreshing", () => {
    const tree = renderer.create(<ListItemContainer refreshing />).toJSON();
    expect(tree).toMatchSnapshot();
})