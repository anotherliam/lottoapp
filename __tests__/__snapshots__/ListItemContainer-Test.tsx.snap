// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and matches snapshot 1`] = `
<View
  refreshing={false}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#eee",
        "elevation": 4,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 12,
        "marginVertical": 8,
        "opacity": 1,
        "padding": 8,
      },
      undefined,
    ]
  }
/>
`;

exports[`renders and matches snapshot when refreshing 1`] = `
<View
  refreshing={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#eee",
        "elevation": 4,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 12,
        "marginVertical": 8,
        "opacity": 0.5,
        "padding": 8,
      },
      undefined,
    ]
  }
/>
`;
